.eleventy.js

```diff
@@ -101,67 +98,65 @@ module.exports = function (eleventyConfig) {
   )
 
   // Code example short codes
-  eleventyConfig.addPairedShortcode(
-    'example',
-    function (
-      code,
-      lang = 'html',
-      stacked,
-      opened,
-      background,
-      themable = true,
-      heighlight,
-      heighlightCssComponent
-    ) {
-      const [codeWebComponent, codeCssComponent] = code
-        .split('<!-- CSS component -->')
-        .map((c) => c.trim())
-      let output = '<docs-example '
-      output += `code="${encodeURIComponent(codeWebComponent)}" `
-
-      if (codeCssComponent) {
-        output += `code-css-component="${encodeURIComponent(
-          codeCssComponent
-        )}" `
-      }
+  eleventyConfig.addPairedShortcode('example', function (code, config) {
+    const defaultConfig = {
+      lang: 'html',
+      stacked: false,
+      opened: false,
+      background: undefined,
+      themable: true,
+      heighlight: undefined,
+      heighlightCssComponent: undefined,
+    }
+    const finalConfig = Object.assign(defaultConfig, JSON.parse(config || '{}'))
```

Comment:
I assume the new eleventy version only supports primitive types for parameters? Otherwise I'd suggest passing an object instead of a stringified JSON object.

Reply:
It's the [liquid template language](https://shopify.github.io/liquid/) where I didn't find any less verbose way of passing an object type prop to the short-code.